From 8f81532599aa59900266f231d107944eef0453d9 Mon Sep 17 00:00:00 2001
From: Aditya Neelamraju <adityanv97@gmail.com>
Date: Tue, 31 Oct 2023 13:40:28 +0000
Subject: [PATCH 2337/2871] clang-format: fix typo in comment

Signed-off-by: Aditya Neelamraju <adityanv97@gmail.com>
Signed-off-by: Junio C Hamano <gitster@pobox.com>
---
 .clang-format | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/.clang-format b/.clang-format
index c592dda681..3ed4fac753 100644
--- a/.clang-format
+++ b/.clang-format
@@ -83,9 +83,9 @@ BinPackParameters: true
 BreakBeforeBraces: Linux
 
 # Break after operators
-# int valuve = aaaaaaaaaaaaa +
-#              bbbbbb -
-#              ccccccccccc;
+# int value = aaaaaaaaaaaaa +
+#             bbbbbb -
+#             ccccccccccc;
 BreakBeforeBinaryOperators: None
 BreakBeforeTernaryOperators: false
 
-- 
2.39.3 (Apple Git-145)

